persist/fs/commitlog/reader.go

```diff
@@ -165,16 +283,10 @@ func (r *reader) readChunk() ([]byte, error) {
 		return nil, err
 	}
 
-	// Extend buffer as necessary
-	if len(r.dataBuffer) < int(size) {
-		diff := int(size) - len(r.dataBuffer)
-		for i := 0; i < diff; i++ {
-			r.dataBuffer = append(r.dataBuffer, 0)
-		}
-	}
+	newBuf := make([]byte, size, size)
```

Comment:
Pretty much. Just add a `BytesPool` in the options (similar to [this](https://github.com/m3db/m3db/blob/master/storage/types.go#L573), initialise with a default in the options ctor like [this](https://github.com/m3db/m3db/blob/master/storage/options.go#L289-L298))